protocol/contracts/amo/helpers/PoolHelper.sol

```diff
@@ -0,0 +1,248 @@
+            revert AMOCommon.InvalidBPSValue(belowTPF);
+        }
+        rebalancePercentageBoundLow = belowTPF;
+        rebalancePercentageBoundUp = aboveTPF;
+    }
+
+    function spotPriceUsingLPRatio() public view returns (uint256 templeBalance, uint256 stableBalance) {
+      uint256[] memory balances = getBalances();
+      if (templeBalancerPoolIndex == 0) {
+          templeBalance = balances[0];
+          stableBalance = balances[1];
+      } else {
+          templeBalance = balances[1];
+          stableBalance = balances[0];
+      }
+    }
+
+    function getSpotPriceScaled() public view returns (uint256 spotPriceScaled) {
+      (uint256 templeBalance, uint256 stableBalance) = spotPriceUsingLPRatio();
+      spotPriceScaled = (TPF_PRECISION * stableBalance) / templeBalance;
+    }
+
+    function isSpotPriceBelowTPF() external view returns (bool) {
+      uint256 spotPriceScaled = getSpotPriceScaled();
+      if (spotPriceScaled < templePriceFloorRatio.numerator) {
+          return true;
+      }
+      return false;
+    }
+
+    // below TPF by a given slippage percentage
+    function isSpotPriceBelowTPF(uint256 slippage) public view returns (bool) {
+      uint256 spotPriceScaled = getSpotPriceScaled();
+      uint256 slippageTPF = (slippage * templePriceFloorRatio.numerator) / templePriceFloorRatio.denominator;
+      if (spotPriceScaled < templePriceFloorRatio.numerator - slippageTPF) {
+          return true;
+      }
+      return false;
+    }
+
+    function isSpotPriceBelowTPFLowerBound() external view returns (bool) {
+        return isSpotPriceBelowTPF(rebalancePercentageBoundLow);
+    }
+
+    function isSpotPriceAboveTPFUpperBound() external view returns (bool) {
+        return isSpotPriceAboveTPF(rebalancePercentageBoundUp);
+    }
+
+    // slippage in bps
+    // above TPF by a given slippage percentage
+    function isSpotPriceAboveTPF(uint256 slippage) public view returns (bool) {
+      uint256 spotPriceScaled = getSpotPriceScaled();
+      uint256 slippageTPF = (slippage * templePriceFloorRatio.numerator) / templePriceFloorRatio.denominator;
+      if (spotPriceScaled > templePriceFloorRatio.numerator + slippageTPF) {
+          return true;
+      }
+      return false;
+    }
+
+     function isSpotPriceAboveTPF() external view returns (bool) {
+        uint256 spotPriceScaled = getSpotPriceScaled();
+        if (spotPriceScaled > templePriceFloorRatio.numerator) {
+            return true;
+        }
+        return false;
+    }
+
+    // @notice will exit take price above tpf by a percentage
+    // percentage in bps
+    // tokensOut: expected min amounts out. for rebalance this is expected Temple tokens out
+    function willExitTakePriceAboveTPFUpperBound(uint256 tokensOut) external view returns (bool) {
+        uint256 percentageIncrease = (templePriceFloorRatio.numerator * rebalancePercentageBoundUp) / templePriceFloorRatio.denominator;
```

Comment:
Given `templePriceFloorRatio.denominator` is fixed, perhaps just change to be the constant `TPF_PRECISION` rather than looking up storage each time?